### ISSUE TITLE
iOS VoiceOver isn't able to go back to SliverAppBar from SliverSafeArea after appbar leaves screen 

### ISSUE THREAD

**Issue by jiahaog** [flutter/flutter#143437]
Internal bug: b/323262670

### Steps to reproduce

1. Launch the app, press the `+` button to launch the page with the SliverAppBar.
2. With VoiceOver on or off (doesn't matter), scroll so that the app bar is out of view.
3. With VoiceOver enabled, scroll back through the elements and try to bring focus to the app bar back button.

After SliverAppBar is no longer visible when VoiceOver is on, VoiceOver can't bring focus to the SliverAppBar.

### Expected results

Voiceover is able to go back to SliverAppBar from SliverSafeArea after app bar leaves screen.

### Actual results

VoiceOver isn't able to go back to SliverAppBar from SliverSafeArea after appbar leaves screen 

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() async {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);
  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title, style: TextStyle(fontFamily: 'ProductSans')),
      ),
      body: Center(
        child: Text(
          'foo.',
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () {
          Navigator.push(
            context,
            MaterialPageRoute(builder: (context) => secondRoute()),
          );
        },
      ),
    );
  }

  Widget secondRoute() {
    return Scaffold(
      body: Stack(children: [
        CustomScrollView(
          slivers: [
            SliverAppBar(actions: []),
            SliverToBoxAdapter(
              child: Text(' tippy'),
            ),
            SliverToBoxAdapter(
              child: Text(' second'),
            ),
            SliverToBoxAdapter(
              child: Text(' third'),
            ),
            const SliverToBoxAdapter(
              child: SizedBox(
                height: 20,
                child: Center(
                  child: Text('Scroll to see the SliverAppBar in effect.'),
                ),
              ),
            ),
            SliverSafeArea(
              sliver: SliverList(
                delegate: SliverChildBuilderDelegate(
                  (BuildContext context, int index) {
                    return Container(
                      color: index.isOdd ? Colors.white : Colors.black12,
                      height: 100.0,
                      child: Center(
                        child: Text('$index',
                            textScaler: const TextScaler.linear(5)),
                      ),
                    );
                  },
                  childCount: 20,
                ),
              ),
            ),
          ],
        ),
      ]),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

This was reproduced on Flutter Master at `c65ab4d513da99419c44efa6172a978631706dec`

**Comment by hangyujin** [comment]
Looks like the semantics tree drops the node of invisible sliver app bar

**Comment by hangyujin** [comment]
Updating the cacheExtent to SliverGeometry of the sliverAppBar will prevent the node to be dropped from semantics tree. 
https://github.com/flutter/flutter/blob/609d30fe5859db3a05c550174fcf7709a33f4888/packages/flutter/lib/src/rendering/sliver_persistent_header.dart#L351


For the provided code sample, not only sliverAppBar, SliverToBoxAdapter widgets are also dropped from the semantics tree, will update them too.    

I also noticed the toolbarOpacity in a11y mode may need to be 1.0.

I am working on a PR for this.
